### ISSUE TITLE
sudden crashes or lags with --corrqueue in a round robin correspondence tournament with phoenixgo

### ISSUE THREAD

**Issue by wonderingabout** [online-go/gtp2ogs#199]
hi

both crashes and issues were fixed by a gtp2ogs restart
i'm using a recent version of gtp2ogs devel (arround the last 10 commits)

so during this tournament (https://online-go.com/tournament/52459) my bot had to play 3 games simultaneously, i used `--corrqueue 1` because gpu cant handle many games (out of vram, 1.6 GB per game, total is 6GB on my gtx 1060), so the games were progressively played simultaneously, one at a time

anyways when game 1 (https://online-go.com/game/20101028) finished, my computer suddenly got really laggy, so i restarted gtp2ogs and it fixed the issue

i noticed that thinking time of the other games doubled after the end of game 1 going from 80sec/move : 
`[19:49] meta-金毛测试-20b: Winrate: 29.999287%, N=27697, Q=-0.400014, p=0.691566, v=-0.554087, cost 77218.007812ms, sims=10008`

to 150sec/move : 
`[19:52] meta-金毛测试-20b: Winrate: 95.522079%, N=6473, Q=0.910442, p=0.421984, v=0.943066, cost 149028.140625ms, sims=10007
`

log of game 1 (not sure what is relevant) : 

```
Oct 26 19:47:03   game.js:490                    [Game 20101030]  Move 150  Got     g11
Oct 26 19:47:03   bot.js:214                     [13764] Calling sendMove with g11
Oct 26 19:47:03   bot.js:214                     [13764] >>> play white g11
Oct 26 19:47:03 !!!!! bot.js:222                     [13764] stderr: 150th move(w): gk, winrate=69.999786%, N=18950, Q=0.399996, p=0.693930, v=0.535022, global_step=639200
Oct 26 19:47:03   bot.js:214                     [13764] <<< =
Oct 26 19:48:17 !!!!! bot.js:222                     [15485] stderr: 175th move(b): dr, winrate=99.331245%, N=8524, Q=0.986625, p=0.757151, v=0.996900, cost 77353.859375ms, sims=10007, height=24, avg_height=11.516632, global_step=639200
Oct 26 19:48:17   bot.js:214                     [15485] <<< = D18
Oct 26 19:48:17   game.js:490                    [Game 20101028]  Move 174  Playing d18 { x: 3, y: 1, text: 'd18', resign: false, pass: false }
Oct 26 19:48:18   connection.js:1391             #  active_game: {"id":20101028,"phase":"play","name":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (20bTurboLz-Elf-v1 vs meta-金毛测试-20b)","player_to_move":625990,"width":19,"height":19,"move_number":175,"paused":0,"private":false,"black":{"username":"meta-金毛测试-20b","id":592558,"rank":36.80948311930709,"professional":false,"accepted":false,"ratings":{"overall":{"deviation":95.45099520611241,"rating":2760.424660297423,"games_played":23,"volatility":0.060090145513783985}}},"white":{"username":"20bTurboLz-Elf-v1","id":625990,"rank":35.112182134606975,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2650.999092339187,"deviation":82.00736778088562,"volatility":0.06084340745467896,"games_played":154}}},"time_per_move":4320}
Oct 26 19:48:18   game.js:490                    [Game 20101028]  Move 174  clock: {"game_id":20101028,"current_player":625990,"black_player_id":592558,"white_player_id":625990,"title":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (20bTurboLz-Elf-v1 vs meta-金毛测试-20b)","last_move":1572112098051,"expiration":1572198487632,"black_time":{"thinking_time":0,"moves_left":12,"block_time":85077.87899999999},"white_time":{"thinking_time":0,"moves_left":13,"block_time":86389.58099999999},"now":1572112098052}
Oct 26 19:48:18   game.js:490                    [Game 20101028]  Move 174  game/20101028/move: { game_id: 20101028, move_number: 175, move: [ 3, 1, 155039 ] }
Oct 26 19:48:18   game.js:490                    [Game 20101028]  Move 175  Ignoring our own move 175
Oct 26 19:48:18   game.js:490                    [Game 20101030]  Move 150  makeMove 150 is 150 !== 150 ?
Oct 26 19:48:18   bot.js:214                     [13764] Generating move for game 20101030
Oct 26 19:48:18   game.js:490                    [Game 20101030]  Move 150  genmove black
Oct 26 19:48:18   bot.js:214                     [13764] >>> genmove black
Oct 26 19:48:19   connection.js:1391             #  active_game: {"id":20101028,"phase":"finished","name":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (20bTurboLz-Elf-v1 vs meta-金毛测试-20b)","player_to_move":625990,"width":19,"height":19,"move_number":175,"paused":0,"private":false,"black":{"username":"meta-金毛测试-20b","id":592558,"rank":36.80948311930709,"professional":false,"accepted":false,"ratings":{"overall":{"deviation":95.45099520611241,"rating":2760.424660297423,"games_played":23,"volatility":0.060090145513783985}}},"white":{"username":"20bTurboLz-Elf-v1","id":625990,"rank":35.112182134606975,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2650.999092339187,"deviation":82.00736778088562,"volatility":0.06084340745467896,"games_played":154}}},"time_per_move":4320}
Oct 26 19:48:19   connection.js:1391             #  20101028 active_game phase === finished
Oct 26 19:48:19   connection.js:174              Starting disconnect Timeout in Connection active_game for 20101028
Oct 26 19:48:19   game.js:490                    [Game 20101028]  Move 175  gamedata       B 20bTurboLz-Elf-v1  [19x19]    
Oct 26 19:48:19   game.js:490                    [Game 20101028]  Move 175  Game over.   Result: B+R      W
Oct 26 19:48:19   bot.js:214                     [15485] Stopping bot
Oct 26 19:48:19   bot.js:214                     [15485] >>> quit
Oct 26 19:48:19   game.js:490                    [Game 20101028]  Move 175  clock: {"game_id":20101028,"current_player":625990,"black_player_id":592558,"white_player_id":625990,"title":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (20bTurboLz-Elf-v1 vs meta-金毛测试-20b)","last_move":1572112098051,"expiration":1572198487632,"black_time":{"thinking_time":0,"moves_left":12,"block_time":85077.87899999999},"white_time":{"thinking_time":0,"moves_left":13,"block_time":86389.58099999999},"now":1572112099728}
Oct 26 19:48:19 !!!!! bot.js:222                     [15485] stderr: *** Aborted at 1572112099 (unix time) try "date -d @1572112099" if you are using GNU date ***
Oct 26 19:48:19 !!!!! bot.js:222                     [15485] stderr: PC: @                0x0 (unknown)
Oct 26 19:48:19   bot.js:214                     [15485] Bot exited
Oct 26 19:48:20   connection.js:1391             #  disconnectFromGame 20101028
Oct 26 19:48:20   game.js:490                    [Game 20101028]  Move 175  Disconnecting from game.
Oct 26 19:48:21   game.js:490                    [Game 20101030]  Move 150  makeMove 150 is 150 !== 150 ?
Oct 26 19:48:21   bot.js:214                     [13764] Generating move for game 20101030
Oct 26 19:48:21   game.js:490                    [Game 20101030]  Move 150  genmove black
Oct 26 19:48:21   bot.js:214                     [13764] >>> genmove black
Oct 26 19:48:24   bot.js:214                     [15485] Killing process directly with a signal
Oct 26 19:49:35 !!!!! bot.js:222                     [13764] stderr: 151th move(b): gl, winrate=29.999287%, N=27697, Q=-0.400014, p=0.691566, v=-0.554087, cost 77218.007812ms, sims=10008, height=32, avg_height=16.704714, global_step=639200
Oct 26 19:49:35   bot.js:214                     [13764] <<< = G12
Oct 26 19:49:35   game.js:490                    [Game 20101030]  Move 150  Playing g12 { x: 6, y: 7, text: 'g12', resign: false, pass: false }
Oct 26 19:49:36   connection.js:1391             #  active_game: {"id":20101030,"phase":"play","name":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (00bSlowLeela vs meta-金毛测试-20b)","player_to_move":640435,"width":19,"height":19,"move_number":151,"paused":0,"private":false,"black":{"username":"meta-金毛测试-20b","id":592558,"rank":36.80948311930709,"professional":false,"accepted":false,"ratings":{"overall":{"deviation":95.45099520611241,"rating":2760.424660297423,"games_played":23,"volatility":0.060090145513783985}}},"white":{"username":"00bSlowLeela","id":640435,"rank":35.27800425979748,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2782.265389172175,"deviation":97.08064998173494,"volatility":0.060371047779368435,"games_played":22}}},"time_per_move":4320}
Oct 26 19:49:36   game.js:490                    [Game 20101030]  Move 150  clock: {"game_id":20101030,"current_player":640435,"black_player_id":592558,"white_player_id":640435,"title":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (00bSlowLeela vs meta-金毛测试-20b)","last_move":1572112175983,"expiration":1572198421574,"black_time":{"thinking_time":0,"moves_left":4,"block_time":82418.127},"white_time":{"thinking_time":0,"moves_left":5,"block_time":86245.59099999997},"now":1572112175984}
Oct 26 19:49:36   game.js:490                    [Game 20101030]  Move 150  game/20101030/move: { game_id: 20101030, move_number: 151, move: [ 6, 7, 152140 ] }
Oct 26 19:49:36   game.js:490                    [Game 20101030]  Move 151  Ignoring our own move 151
Oct 26 19:49:36   game.js:490                    [Game 20101026]  Move 123  makeMove 123 is 123 !== 123 ?
Oct 26 19:49:36   bot.js:214                     [18245] Generating move for game 20101026
Oct 26 19:49:36   game.js:490                    [Game 20101026]  Move 123  genmove white
Oct 26 19:49:36   bot.js:214                     [18245] >>> genmove white
Oct 26 19:49:52   connection.js:1391             #  active_game: {"id":20101030,"phase":"play","name":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (00bSlowLeela vs meta-金毛测试-20b)","player_to_move":592558,"width":19,"height":19,"move_number":152,"paused":0,"private":false,"black":{"username":"meta-金毛测试-20b","id":592558,"rank":36.80948311930709,"professional":false,"accepted":false,"ratings":{"overall":{"deviation":95.45099520611241,"rating":2760.424660297423,"games_played":23,"volatility":0.060090145513783985}}},"white":{"username":"00bSlowLeela","id":640435,"rank":35.27800425979748,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2782.265389172175,"deviation":97.08064998173494,"volatility":0.060371047779368435,"games_played":22}}},"time_per_move":4320}
Oct 26 19:49:52   connection.js:1391             #  Connected to game 20101030 already
Oct 26 19:49:52   game.js:490                    [Game 20101030]  Move 151  clock: {"game_id":20101030,"current_player":592558,"black_player_id":592558,"white_player_id":640435,"title":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (00bSlowLeela vs meta-金毛测试-20b)","last_move":1572112192149,"expiration":1572194610276,"black_time":{"thinking_time":0,"moves_left":4,"block_time":82418.127},"white_time":{"thinking_time":0,"moves_left":4,"block_time":86229.42499999997},"now":1572112192150}
Oct 26 19:49:52   game.js:490                    [Game 20101030]  Move 151  game/20101030/move: { game_id: 20101030, move_number: 152, move: [ 7, 8, 16166 ] }
Oct 26 19:49:52   game.js:490                    [Game 20101030]  Move 152  Got     h11
Oct 26 19:49:52   bot.js:214                     [13764] Calling sendMove with h11
Oct 26 19:49:52   bot.js:214                     [13764] >>> play white h11
Oct 26 19:52:04 !!!!! bot.js:222                     [13764] stderr: 153th move(b): hk, winrate=95.522079%, N=6473, Q=0.910442, p=0.421984, v=0.943066, cost 149028.140625ms, sims=10007, height=20, avg_height=8.043097, global_step=639200
Oct 26 19:52:04   bot.js:214                     [13764] <<< = H11
Oct 26 19:52:04   game.js:490                    [Game 20101030]  Move 152  Playing h11 { x: 7, y: 8, text: 'h11', resign: false, pass: false }
Oct 26 19:52:04 !!!!! bot.js:222                     [13764] stderr: F1026 19:52:04.955504 13764 mcts_engine.cc:154] Check failed: ret == 0 (-1 vs. 0) Move: failed, hk, ret-1
Oct 26 19:52:04 !!!!! bot.js:222                     [13764] stderr: *** Check failure stack trace: ***
Oct 26 19:52:04 !!!!! bot.js:222                     [13764] stderr: *** Aborted at 1572112324 (unix time) try "date -d @1572112324" if you are using GNU date ***
Oct 26 19:52:04 !!!!! bot.js:222                     [13764] stderr: PC: @                0x0 (unknown)
Oct 26 19:52:05   bot.js:214                     [13764] Bot exited
Oct 26 19:52:05 !!!!! bot.js:222                     [18245] stderr: 124th move(w): lp, winrate=57.043121%, N=6175, Q=0.140862, p=0.550667, v=0.414196, cost 149031.812500ms, sims=10008, height=29, avg_height=11.145996, global_step=639200
Oct 26 19:52:05   bot.js:214                     [18245] <<< = M16
Oct 26 19:52:05   game.js:490                    [Game 20101026]  Move 123  Playing m16 { x: 11, y: 3, text: 'm16', resign: false, pass: false }
Oct 26 19:52:05   game.js:490                    [Game 20101026]  Move 123  clock: {"game_id":20101026,"current_player":602660,"black_player_id":602660,"white_player_id":592558,"title":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (meta-金毛测试-20b vs 15bTurboLeela)","last_move":1572112325867,"expiration":1572198723580,"black_time":{"thinking_time":0,"moves_left":18,"block_time":86397.713},"white_time":{"thinking_time":0,"moves_left":18,"block_time":85245.08600000001},"now":1572112325868}
Oct 26 19:52:05   game.js:490                    [Game 20101026]  Move 123  game/20101026/move: { game_id: 20101026, move_number: 124, move: [ 11, 3, 923165 ] }
Oct 26 19:52:05   game.js:490                    [Game 20101026]  Move 124  Ignoring our own move 124
Oct 26 19:52:05   connection.js:1391             #  active_game: {"id":20101026,"phase":"play","name":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (meta-金毛测试-20b vs 15bTurboLeela)","player_to_move":602660,"width":19,"height":19,"move_number":124,"paused":0,"private":false,"black":{"username":"15bTurboLeela","id":602660,"rank":35.56195510977517,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2677.522241926865,"deviation":86.55796868278182,"volatility":0.05989793305732274,"games_played":727}}},"white":{"username":"meta-金毛测试-20b","id":592558,"rank":36.80948311930709,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2784.953901141961,"deviation":111.81904248383128,"volatility":0.06008247264443957,"games_played":24}}},"time_per_move":4320}
Oct 26 19:52:07   connection.js:1391             #  active_game: {"id":20101026,"phase":"play","name":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (meta-金毛测试-20b vs 15bTurboLeela)","player_to_move":592558,"width":19,"height":19,"move_number":125,"paused":0,"private":false,"black":{"username":"15bTurboLeela","id":602660,"rank":35.56195510977517,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2677.522241926865,"deviation":86.55796868278182,"volatility":0.05989793305732274,"games_played":727}}},"white":{"username":"meta-金毛测试-20b","id":592558,"rank":36.80948311930709,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2784.953901141961,"deviation":111.81904248383128,"volatility":0.06008247264443957,"games_played":24}}},"time_per_move":4320}
Oct 26 19:52:07   connection.js:1391             #  Connected to game 20101026 already
Oct 26 19:52:07   game.js:490                    [Game 20101026]  Move 124  clock: {"game_id":20101026,"current_player":592558,"black_player_id":602660,"white_player_id":592558,"title":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (meta-金毛测试-20b vs 15bTurboLeela)","last_move":1572112327543,"expiration":1572197572629,"black_time":{"thinking_time":0,"moves_left":17,"block_time":86396.037},"white_time":{"thinking_time":0,"moves_left":18,"block_time":85245.08600000001},"now":1572112327543}
Oct 26 19:52:07   game.js:490                    [Game 20101026]  Move 124  game/20101026/move: { game_id: 20101026, move_number: 125, move: [ 11, 4, 1676 ] }
Oct 26 19:52:07   game.js:490                    [Game 20101026]  Move 125  Got     m15
Oct 26 19:52:07   bot.js:214                     [18245] Calling sendMove with m15
Oct 26 19:52:07   bot.js:214                     [18245] >>> play black m15
Oct 26 19:52:07   game.js:490                    [Game 20101026]  Move 125  makeMove 125 is 125 !== 125 ?
Oct 26 19:52:07   bot.js:214                     [18245] Generating move for game 20101026
Oct 26 19:52:07   game.js:490                    [Game 20101026]  Move 125  genmove white
Oct 26 19:52:07   bot.js:214                     [18245] >>> genmove white
Oct 26 19:52:07 !!!!! bot.js:222                     [18245] stderr: 125th move(b): lo, winrate=43.842934%, N=5656, Q=-0.123141, p=0.433415, v=-0.462517, global_step=639200
Oct 26 19:52:07   bot.js:214                     [18245] <<< =
^C
```

i thought maybe its a phoenixgo specific issue, so i thought nothing of it, but then at the end of game 2 (https://online-go.com/game/20101030) something wrong happened again in game 3 (https://online-go.com/game/20101026) : 
this time my bot winrate suddenly increased from 60% to 98% and then it crashed immediately without playing the move (move 162).
Such a winrate change had no reason to happen in that context, so i'm assuming after game 2 ended either the colors got switched, or maybe a pass was played on the gtp console of phoenixgo, or something like that, but anyways the bot crashed again at the end of game 2 as you can see in the logs : 

```
Oct 26 20:32:10   bot.js:214                     [23313] <<< =
Oct 26 20:33:26 !!!!! bot.js:222                     [21684] stderr: 178th move(w): qa, winrate=84.027885%, N=5094, Q=0.680558, p=0.690361, v=0.373733, global_step=639200
Oct 26 20:33:26   bot.js:214                     [21684] <<< = resign
Oct 26 20:33:26   game.js:490                    [Game 20101030]  Move 178  Resigning
Oct 26 20:33:26   game.js:490                    [Game 20101026]  Move 159  makeMove 159 is 159 !== 159 ?
Oct 26 20:33:26   bot.js:214                     [23313] Generating move for game 20101026
Oct 26 20:33:26   game.js:490                    [Game 20101026]  Move 159  genmove white
Oct 26 20:33:26   bot.js:214                     [23313] >>> genmove white
Oct 26 20:33:26   connection.js:1391             #  active_game: {"id":20101030,"phase":"finished","name":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (00bSlowLeela vs meta-金毛测试-20b)","player_to_move":592558,"width":19,"height":19,"move_number":178,"paused":0,"private":false,"black":{"username":"meta-金毛测试-20b","id":592558,"rank":36.80948311930709,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2784.953901141961,"deviation":111.81904248383128,"volatility":0.06008247264443957,"games_played":24}}},"white":{"username":"00bSlowLeela","id":640435,"rank":35.27800425979748,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2782.265389172175,"deviation":97.08064998173494,"volatility":0.060371047779368435,"games_played":22}}},"time_per_move":4320}
Oct 26 20:33:26   connection.js:1391             #  20101030 active_game phase === finished
Oct 26 20:33:26   connection.js:174              Starting disconnect Timeout in Connection active_game for 20101030
Oct 26 20:33:26   game.js:490                    [Game 20101030]  Move 178  gamedata       B 00bSlowLeela  [19x19]    
Oct 26 20:33:26   game.js:490                    [Game 20101030]  Move 178  Game over.   Result: W+R         L
Oct 26 20:33:26   bot.js:214                     [21684] Stopping bot
Oct 26 20:33:26   bot.js:214                     [21684] >>> quit
Oct 26 20:33:26   game.js:490                    [Game 20101030]  Move 178  clock: {"game_id":20101030,"current_player":592558,"black_player_id":592558,"white_player_id":640435,"title":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (00bSlowLeela vs meta-金毛测试-20b)","last_move":1572114657693,"expiration":1572199487821,"black_time":{"thinking_time":0,"moves_left":11,"block_time":84830.12799999998},"white_time":{"thinking_time":0,"moves_left":11,"block_time":86328.736},"now":1572114806291}
Oct 26 20:33:26 !!!!! bot.js:222                     [21684] stderr: *** Aborted at 1572114806 (unix time) try "date -d @1572114806" if you are using GNU date ***
Oct 26 20:33:26 !!!!! bot.js:222                     [21684] stderr: PC: @                0x0 (unknown)
Oct 26 20:33:26   bot.js:214                     [21684] Bot exited
Oct 26 20:33:27   connection.js:1391             #  disconnectFromGame 20101030
Oct 26 20:33:27   game.js:490                    [Game 20101030]  Move 178  Disconnecting from game.
Oct 26 20:33:28   game.js:490                    [Game 20101026]  Move 159  makeMove 159 is 159 !== 159 ?
Oct 26 20:33:28   bot.js:214                     [23313] Generating move for game 20101026
Oct 26 20:33:28   game.js:490                    [Game 20101026]  Move 159  genmove white
Oct 26 20:33:28   bot.js:214                     [23313] >>> genmove white
Oct 26 20:33:31   bot.js:214                     [21684] Killing process directly with a signal
Oct 26 20:34:44 !!!!! bot.js:222                     [23313] stderr: 160th move(w): am, winrate=60.715931%, N=8887, Q=0.214319, p=0.354375, v=0.415143, cost 78154.921875ms, sims=10008, height=29, avg_height=13.047717, global_step=639200
Oct 26 20:34:44   bot.js:214                     [23313] <<< = A13
Oct 26 20:34:44   game.js:490                    [Game 20101026]  Move 159  Playing a13 { x: 0, y: 6, text: 'a13', resign: false, pass: false }
Oct 26 20:34:44   connection.js:1391             #  active_game: {"id":20101026,"phase":"play","name":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (meta-金毛测试-20b vs 15bTurboLeela)","player_to_move":602660,"width":19,"height":19,"move_number":160,"paused":0,"private":false,"black":{"username":"15bTurboLeela","id":602660,"rank":35.56195510977517,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2701.787531952058,"deviation":84.25093401294474,"volatility":0.0599405951603354,"games_played":728}}},"white":{"username":"meta-金毛测试-20b","id":592558,"rank":36.80948311930709,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2784.953901141961,"deviation":111.81904248383128,"volatility":0.06008247264443957,"games_played":24}}},"time_per_move":4320}
Oct 26 20:34:44   game.js:490                    [Game 20101026]  Move 159  clock: {"game_id":20101026,"current_player":602660,"black_player_id":602660,"white_player_id":592558,"title":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (meta-金毛测试-20b vs 15bTurboLeela)","last_move":1572114884496,"expiration":1572201284496,"black_time":{"thinking_time":0,"moves_left":20,"block_time":86400},"white_time":{"thinking_time":0,"moves_left":20,"block_time":86400},"now":1572114884496}
Oct 26 20:34:44   game.js:490                    [Game 20101026]  Move 159  game/20101026/move: { game_id: 20101026, move_number: 160, move: [ 0, 6, 154338 ] }
Oct 26 20:34:44   game.js:490                    [Game 20101026]  Move 160  Ignoring our own move 160
Oct 26 20:34:47   connection.js:1391             #  active_game: {"id":20101026,"phase":"play","name":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (meta-金毛测试-20b vs 15bTurboLeela)","player_to_move":592558,"width":19,"height":19,"move_number":161,"paused":0,"private":false,"black":{"username":"15bTurboLeela","id":602660,"rank":35.56195510977517,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2701.787531952058,"deviation":84.25093401294474,"volatility":0.0599405951603354,"games_played":728}}},"white":{"username":"meta-金毛测试-20b","id":592558,"rank":36.80948311930709,"professional":false,"accepted":false,"ratings":{"overall":{"rating":2784.953901141961,"deviation":111.81904248383128,"volatility":0.06008247264443957,"games_played":24}}},"time_per_move":4320}
Oct 26 20:34:47   connection.js:1391             #  Connected to game 20101026 already
Oct 26 20:34:47   game.js:490                    [Game 20101026]  Move 160  clock: {"game_id":20101026,"current_player":592558,"black_player_id":602660,"white_player_id":592558,"title":"Tournament Game: metaBot AI Tournament OGS 2019x19 v2 (52459) R:1 (meta-金毛测试-20b vs 15bTurboLeela)","last_move":1572114887505,"expiration":1572201287505,"black_time":{"thinking_time":0,"moves_left":19,"block_time":86396.991},"white_time":{"thinking_time":0,"moves_left":20,"block_time":86400},"now":1572114887506}
Oct 26 20:34:47   game.js:490                    [Game 20101026]  Move 160  game/20101026/move: { game_id: 20101026, move_number: 161, move: [ 3, 13, 3009 ] }
Oct 26 20:34:47   game.js:490                    [Game 20101026]  Move 161  Got     d6
Oct 26 20:34:47   bot.js:214                     [23313] Calling sendMove with d6
Oct 26 20:34:47   bot.js:214                     [23313] >>> play black d6
Oct 26 20:34:47   game.js:490                    [Game 20101026]  Move 161  makeMove 161 is 161 !== 161 ?
Oct 26 20:34:47   bot.js:214                     [23313] Generating move for game 20101026
Oct 26 20:34:47   game.js:490                    [Game 20101026]  Move 161  genmove white
Oct 26 20:34:47   bot.js:214                     [23313] >>> genmove white
Oct 26 20:36:02 !!!!! bot.js:222                     [23313] stderr: 162th move(w): df, winrate=98.948624%, N=6557, Q=0.978972, p=0.151510, v=0.977798, cost 77690.320312ms, sims=10008, height=17, avg_height=7.859396, global_step=639200
Oct 26 20:36:02   bot.js:214                     [23313] <<< = D6
Oct 26 20:36:02   game.js:490                    [Game 20101026]  Move 161  Playing d6 { x: 3, y: 13, text: 'd6', resign: false, pass: false }
Oct 26 20:36:02 !!!!! bot.js:222                     [23313] stderr: F1026 20:36:02.125913 23313 mcts_engine.cc:154] Check failed: ret == 0 (-1 vs. 0) Move: failed, df, ret-1
Oct 26 20:36:02 !!!!! bot.js:222                     [23313] stderr: *** Check failure stack trace: ***
Oct 26 20:36:02 !!!!! bot.js:222                     [23313] stderr: *** Aborted at 1572114962 (unix time) try "date -d @1572114962" if you are using GNU date ***
Oct 26 20:36:02 !!!!! bot.js:222                     [23313] stderr: PC: @                0x0 (unknown)
Oct 26 20:36:02   bot.js:214                     [23313] Bot exited
Oct 26 20:37:49   connection.js:1391             #  Dumping status of all connected games
Oct 26 20:37:49   connection.js:1391             #  game_id=20101026: black=15bTurboLeela white=meta-金毛测试-20b bot_turn idle_time=181.579s bot.proc.pid=23313 bot.dead=true bot.failed=false
Oct 26 20:37:49   connection.js:1391             #  Dump complete
^C
```

fixed again by a gtp2ogs restart

so maybe its all phoenixgo specific, but maybe gtp2ogs can improve some things in `--corrqueue` behaviour to make it more universal, like for example, reset the corrqueue and restart the bot if it crashes ? (i know it shoud work fine with leela zero)

thanks !

**Comment by roy7** [comment]
Did you update your gtp2ogs since the https://github.com/online-go/gtp2ogs/pull/197 patch was applied?

It seems your game state got out of sync, black already moved to d6 at move 161 but then when we 'genmove white' your bot tried to play d6 as well, which the server rejected as an invalid move I assume. But there's also the crash from the bot with the same error message from your prior bug report.

However I don't see you getting extra 'gamedata' packets like before so that wouldn't be the cause, but maybe you didn't run with --debug?

My first guess is maybe your gtp2ogs is slightly out of date and is missing the prior bug fix?

**Comment by roy7** [comment]
The prior bug I'm referring to is https://github.com/online-go/gtp2ogs/issues/190#issuecomment-513562453. If the logs were from --debug we'd know for sure if there was an extra gamedata packet in the middle there.

**Comment by wonderingabout** [comment]
the branch i'm using should be this one : https://github.com/wonderingabout/gtp2ogs/tree/roy7-test-beta

also my full bot command was : 

```sudo nvidia-smi -pm 1 && sudo nvidia-smi -pl 70 && nvidia-smi -i 0 -q -d POWER,CLOCK && cd ~/PhoenixGo/etc && sudo nano mcts_1gpu.conf && sudo node /usr/lib/node_modules/gtp2ogs/gtp2ogs.js --apikey meow --username meta-金毛测试-20b --hidden --debug --maxconnectedgamesperuser 1 --maxconnectedgames 1 --corrqueue --boardsizes 19 --noclock --persist --nopause --fakerank 7d --noautohandicap --maxhandicap 0 --komis 7.5,automatic --minrank 1d --unrankedonly --timecontrols fischer,byoyomi --speeds live --minmaintimelive 300 --maxmaintimelive 600 --minperiodtimelive 60 --maxperiodtimelive 180 --maxperiodslive 5 --greeting "hi, now focused on AI vs AI games, this bot uses the original native phoenixgo engine provided by tencent https://github.com/Tencent/PhoenixGo 20 blocks version with default settings (batch size 4 , 8 threads), not leela zero, have fun :) , if you want to have a bot on OGS, you can follow this tutorial : https://github.com/wonderingabout/gtp2ogs-tutorial " --farewell "thank you for the game, hope you enjoyed and/or learned :) there is victory in defeat :) if you want my bot to play a custom opening position against any bot, send it to me here (forked games) https://online-go.com/review/363869 , if you want to have a bot on OGS, you can follow this tutorial : https://github.com/wonderingabout/gtp2ogs-tutorial " --rejectnew --rejectmsg "playing only bot vs bot games, sorry" -- /home/amd2019/PhoenixGo/bazel-bin/mcts/mcts_main --gtp --config_path=/home/amd2019/PhoenixGo/etc/mcts_1gpu.conf```

**Comment by wonderingabout** [comment]
so i was using debug

i guess i have to update my gtp2ogs then :+1: 
would that fix the other issue as well ? (because the 2 crsahes seem to be different)

**Comment by roy7** [comment]
If you looked at the PR I linked it was only one line of code. See if that change is in your version

**Comment by wonderingabout** [comment]
i remember the PR, at that time i was just lazy to update my gtp2ogs :sweat_smile: 
but i wanted to do it some time recently anyways because of the greet at game connection commit

and indeed my branch doesnt have the fix https://github.com/wonderingabout/gtp2ogs/blob/roy7-test-beta/game.js#L221 and other things as well it seems
anyways going to add the fix to my branch for now :)

**Comment by wonderingabout** [comment]
then would that fix address both issues, or is it different problems ? (one is a crash, the other one is a continuous lag with double thinking time)

**Comment by roy7** [comment]
I have no idea on the thinking time. But you can at least fix the one problem and see. It is possible after the crash and restart it started thinking immediately in spite of the corrqueue setting. So you had 2 running at once after the crash.

**Comment by wonderingabout** [comment]
i see, because to multiply by x the thinking time the only way would be to increase the number of simultaneous games
anyways yes for now i'll just update gtp2ogs and we'll see for next tournament :)
i needed to do it anyways, thanks :1st_place_medal:

**Comment by roy7** [comment]
@wonderingabout Would you like to try another corr tournament sometime and see if you can reproduce any issues, or if not, go and and close this?

**Comment by wonderingabout** [comment]
>i thought maybe its a phoenixgo specific issue, so i thought nothing of it, but then at the end of game 2 (https://online-go.com/game/20101030) something wrong happened again in game 3 (https://online-go.com/game/20101026) :
this time my bot winrate suddenly increased from 60% to 98% and then it crashed immediately without playing the move (move 162).

this comment i wrote in the past makes me think we may be in the case of extra gamedata packet sent causing a second genmove, so this would be same as #256 

so i think the gamedata fix in #296 and #300 fixed it now

if not, we should open a new issue and look at it again, the old codebase is way to old to debug this now :)
